udata/features/territories/__init__.py

```diff
@@ -20,3 +24,37 @@ def check_for_territories(query):
         qs = qs(db.Q(name__istartswith=query) | db.Q(name__iexact=query))
     # Sort matching results by population and area.
     return qs.order_by('-population', '-area')
+
+
+def check_for_counties(query):
+    """
+    Return a geozone queryset of counties given the `query`.
+
+    If it's a 2-digits code try it, otherwise use the name.
+    """
+    if (not query or len(query) < 2 or
+            not current_app.config.get('ACTIVATE_TERRITORIES')):
+        return GeoZone.objects.none()
+    qs = GeoZone.objects(level='fr/county')
+    if len(query) == 2 and query.isdigit():
+        # Check county by code.
+        qs = qs(db.Q(code=query))
+    else:
+        # Check names starting with query or exact match.
+        qs = qs(db.Q(name__istartswith=query) | db.Q(name__iexact=query))
+    # Sort matching results by population and area.
+    return qs.order_by('-population', '-area')
+
+
+def check_for_regions(query):
+    """
+    Return a geozone queryset of regions given the `query`.
+    """
+    if (not query or len(query) < 4 or
+            not current_app.config.get('ACTIVATE_TERRITORIES')):
+        return GeoZone.objects.none()
+    qs = GeoZone.objects(level='fr/region')
+    # Check names starting with query or exact match.
+    qs = qs(db.Q(name__istartswith=query) | db.Q(name__iexact=query))
+    # Sort matching results by population and area.
+    return qs.order_by('-population', '-area')
```

Comment:
My point was the opposite: always query both name and code :)
I'm a suburban ;)